### PR TITLE
Add a test that we don't get a "Server" header

### PR DESCRIPTION
**Pull request by digitalronin**
Normally, I'd put this test in a spec of its own,
but these tests take so long to set up that it
makes more sense to have multiple expectations in
a single spec.

### CODE REVIEW

**Review by mogaal — APPROVED** [review]

**Comment by digitalronin** [comment]
depends on https://github.com/ministryofjustice/cloud-platform-infrastructure/pull/955, so I'll merge this after that is merged and applied

**Pull request merged**
